src/openforms/tests/e2e/input_validation_base.py

```diff
@@ -104,16 +108,24 @@ async def _assertFrontendValidation(
             await page.goto(url)
             await page.get_by_role("button", name="Formulier starten").click()
 
-            # fill in the test input
-            await self._locate_input(page, label).fill(ui_input)
+            # fill in the test input in case of text otherwise continue
+            if input_type == "text":
+                await self._locate_input(page, label).fill(ui_input)
```

Comment:
I'm not a fan of this, I think it's better to wrap this in its own private method that can be overridden on a test case basis so that you can implement the UI interaction specifically for the component type being tested. E.g. for a select, you would need to do some aria-role comobox shenanigans and for checkboxes it would be a `.click` or maybe some control to check/uncheck it (not sure what Playwright offers here).